hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -382,6 +383,12 @@ SqlResult execute(ShowStatementPlan plan) {
             case TYPES:
                 rows = catalog.getTypeNames().stream();
                 break;
+            case DATALINKS:
+                InternalDataLinkService service = nodeEngine.getDataLinkService();
+                assert service instanceof DataLinkServiceImpl;
```

Comment:
I do not like the "assert before cast" pattern
```suggestion
```

Reply:
I wonder : why?

Reply:
It does not add value, JVM does the check nevertheless. And it is an additional line of code, which is just noise when reading it (and you read the code many more times than writing it). Some asserts _are_ useful, so I'd rather not train ourselves to skip them when reading the code.